tests/general/IndentationBearTest.py

```diff
@@ -0,0 +1,155 @@
+        self.verify_bear(valid_file1)
+
+        valid_file2 =\
+            ('R("strings can span\n',
+             'multiple lines as well{")\n'
+             'but the bear works correctly\n')
+        self.verify_bear(valid_file2)
+
+        valid_file3 =\
+            ('"this should indent"{ "hopefully"\n',
+             '\tand it does\n',
+             '}\n')
+        self.verify_bear(valid_file3)
+
+    def test_within_comments(self):
+        valid_file1 =\
+            ('//indent within comments{\n',
+             'remains unindented\n')
+        self.verify_bear(valid_file1)
+
+        valid_file2 =\
+            ('/*Indent within\n',
+             'lines of multiline comment {\n',
+             'doesnt have any effect{ */\n',
+             'no affect on regular lines as well\n')
+        self.verify_bear(valid_file2)
+
+        valid_file3 =\
+            ('/*this should indent*/{ /*hopefully*/\n',
+             '\tand it does\n',
+             '}\n')
+        self.verify_bear(valid_file3)
+
+    def test_branch_indents(self):
+        valid_file =\
+            ('branch indents{\n',
+             '\tsecond branch{\n',
+             '\t\twithin second branch\n',
+             '\t}\n',
+             '}\n',)
+        self.verify_bear(valid_file)
+
+    def test_bracket_matching(self):
+        valid_file = ("{{{}{}}",
+                      "\tone_indent",
+                      "}")
+        invalid_file = ("{{{}{}}",
+                        "didn't give indent",
+                        "}")
+        self.verify_bear(valid_file, invalid_file)
+
+    def test_blank_lines(self):
+        valid_file = ("{ trying indent",
+                      "    ",
+                      "\tIndents even after blank line}")
+        invalid_file = ("{ trying indent",
+                        "    ",
+                        "should've Indented after blank line}")
+        self.verify_bear(valid_file, invalid_file)
+
+    def test_settings(self):
+        section = Section("")
+        section.append(Setting('language', 'c'))
+        section.append(Setting('language_family', 'c'))
+        section.append(Setting('use_spaces', True))
+        section.append(Setting('tab_width', 6))
+        valid_file = ('{\n',
+                      # Start ignoring SpaceConsistencyBear
+                      '      6 spaces of indentation\n'
+                      # Stop ignoring
+                      '}\n')
+
+        invalid_file = ('{\n',
+                        # Start ignoring SpaceConsistencyBearW
+                        '    4 spaces of indentation\n'
+                        # Stop ignoring
+                        '}\n')
+        self.verify_bear(valid_file, invalid_file, section)
+
+    def test_unmatched_indents(self):
+        valid_file = ('{}\n',)
+        invalid_file = ('{\n',)
+        self.verify_bear(valid_file, invalid_file)
+
+        invalid_file2 = ('{}}\n',)
+        self.verify_bear(valid_file=None, invalid_file=invalid_file2)
```

Comment:
i tried with the files of the git repository , some with 2000-3000 lines, so the basic problem is the keywords but other than it works!